src/Symfony/Component/BrowserKit/Client.php

```diff
@@ -231,11 +231,11 @@ public function click(Link $link)
      *
      * @return Crawler
      */
-    public function submit(Form $form, array $values = array())
+    public function submit(Form $form, array $values = array(), $serverParameters = array())
```

Comment:
Instead of allowing to pass any server parameters, I think we should restrict to only HTTP headers. What do you think?

Reply:
well, that's not consistent with the `request` API, which expects to get things in the `$_SERVER` format. I think consistency is better for now.

Reply:
Not sure I understand consistency here. Changing anything besides HTTP headers when submitting a form should not be allowed.

Reply:
I mean, consistency between the different APIs of BrowserKit (namely `request` and `submit` here)

Reply:
I created the Pull Request to Mink BrowserKit to show the exakt usage for my usecase. Therefor I might leave it as it is. If you would prefer to filter parameters i can implement that as well